lib/_http_client.js

```diff
@@ -197,7 +197,14 @@ function ClientRequest(input, options, cb) {
   }
   this.insecureHTTPParser = insecureHTTPParser;
 
-  this.path = options.path || '/';
+  this.path = (options.path && method === 'CONNECT' &&
+    options.path.charAt(0) === '/' ?
+    options.path.slice(1) : options.path) || '/';
```

Comment:
This is much cleaner:

```je
  if (options.path) {
    if (method === 'CONNECT') {
      if (options.path.charAt(0) === '/') {
        this.path = options.path.slice(1);
      } else {
        this.path = options.path;
      }
    } else {
      this.path = options.path;
    }
  } else {
    this.path = '/';
  }
```

Reply:
@bnoordhuis Do you mean this?

```js
  const {path} = options;
  if (path) {
    if (method === 'CONNECT') {
      if (path.charAt(0) === '/') {
        this.path = path.slice(1);
      } else {
        this.path = path;
      }
    } else {
      this.path = path;
    }
  } else {
    this.path = '/';
  }
```

Reply:
> Agreed, and as a micro-optimization: assign to a `let path` first and then do `this.path = path;` outside the `if` block.
> 
> (Ask me why if you want details but it has to do with how V8 stores object properties.)

Yes, I was about to ask the reason for this, could you elaborate as why we are avoiding the direct use of `options.path`?

>   const {path} = options;
There is already a `let path` variable declared in `_http_client.js`. I think it must be reused.